pool/src/lib.rs

```diff
@@ -553,17 +596,25 @@ impl LendingPool {
             return Ok(false);
         }
 
-        let collat_coeff = Self::get_collateral_coeff(env, reserve)?;
+        let collat_coeff = Self::get_collat_coeff(env, &asset, reserve)?;
         let underlying_asset = token::Client::new(env, asset);
         let s_token = STokenClient::new(env, &reserve.s_token_address);
-        let is_first_deposit = s_token.balance(who) == 0;
 
+        let s_token_supply = s_token.total_supply();
```

Comment:
Must it be replaced with the `read_stoken_supply` since it implies that they both returns the same value.